File: sdks/java/src/test/java/dev/sequins/otel/SequinsTest.java
package dev.sequins.otel;

import io.opentelemetry.api.trace.Tracer;
import io.opentelemetry.api.metrics.Meter;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class SequinsTest {

    @Test
    void initReturnsHandle() {
        Sequins sequins = Sequins.init("test-service");
        assertNotNull(sequins);
        sequins.close();
    }

    @Test
    void tracerProviderIsNotNull() {
        Sequins sequins = Sequins.init("test-tracer");
        assertNotNull(sequins.getTracerProvider());
        sequins.close();
    }

    @Test
    void meterProviderIsNotNull() {
        Sequins sequins = Sequins.init("test-meter");
        assertNotNull(sequins.getMeterProvider());
        sequins.close();
    }

    @Test
    void loggerProviderIsNotNull() {
        Sequins sequins = Sequins.init("test-logger");
        assertNotNull(sequins.getLoggerProvider());
        sequins.close();
    }

    @Test
    void canGetTracerFromProvider() {
        Sequins sequins = Sequins.init("test-get-tracer");
        Tracer tracer = sequins.getTracerProvider().tracerBuilder("my-lib").build();
        assertNotNull(tracer);
        sequins.close();
    }

    @Test
    void canGetMeterFromProvider() {
        Sequins sequins = Sequins.init("test-get-meter");
        Meter meter = sequins.getMeterProvider().meterBuilder("my-lib").build();
        assertNotNull(meter);
        sequins.close();
    }

    @Test
    void closeDoesNotThrow() {
        Sequins sequins = Sequins.init("test-close");
        assertDoesNotThrow(sequins::close);
    }

    @Test
    void customConfig() {
        SequinsConfig config = SequinsConfig.builder()
                .serviceName("custom-service")
                .endpoint("http://localhost:4317")
                .build();
        Sequins sequins = Sequins.init(config);
        assertNotNull(sequins);
        sequins.close();
    }
}

File: sdks/go/sequins_test.go
package sequins_test

import (
	"context"
	"testing"

	sequins "github.com/sequins-dev/otel-go"
)

func TestInit_ReturnsHandle(t *testing.T) {
	ctx := context.Background()
	s, err := sequins.Init(ctx, sequins.Config{
		ServiceName: "test-service",
		Endpoint:    "localhost:4317",
	})
	if err != nil {
		t.Fatalf("Init returned error: %v", err)
	}
	if s.TracerProvider == nil {
		t.Error("TracerProvider is nil")
	}
	if s.MeterProvider == nil {
		t.Error("MeterProvider is nil")
	}
	if s.LoggerProvider == nil {
		t.Error("LoggerProvider is nil")
	}
	if err := s.Shutdown(ctx); err != nil {
		t.Errorf("Shutdown returned error: %v", err)
	}
}

func TestInit_TracerWorks(t *testing.T) {
	ctx := context.Background()
	s, err := sequins.Init(ctx, sequins.Config{ServiceName: "test-tracer"})
	if err != nil {
		t.Fatalf("Init returned error: %v", err)
	}
	tracer := s.Tracer("my-lib")
	if tracer == nil {
		t.Error("Tracer is nil")
	}
	_ = s.Shutdown(ctx)
}

func TestInit_MeterWorks(t *testing.T) {
	ctx := context.Background()
	s, err := sequins.Init(ctx, sequins.Config{ServiceName: "test-meter"})
	if err != nil {
		t.Fatalf("Init returned error: %v", err)
	}
	meter := s.Meter("my-lib")
	if meter == nil {
		t.Error("Meter is nil")
	}
	_ = s.Shutdown(ctx)
}

func TestInit_DefaultServiceName(t *testing.T) {
	ctx := context.Background()
	s, err := sequins.Init(ctx, sequins.Config{})
	if err != nil {
		t.Fatalf("Init returned error: %v", err)
	}
	_ = s.Shutdown(ctx)
}
